test/Microsoft.DocAsCode.Tests.Common/TestLoggerListener.cs

```diff
@@ -15,22 +16,28 @@ public class TestLoggerListener : ILoggerListener
 
         public LogLevel LogLevelThreshold { get; set; }
 
-        public TestLoggerListener(string phase = null)
-        {
-            Phase = phase;
-        }
+        public Func<ILogItem, bool> PhaseMatcher { get; set; }
 
-        public void Dispose()
+        public TestLoggerListener(string phase = null,
+            LogLevel logLevelThreshold = LogLevel.Warning,
+            Func<ILogItem, bool> phaseMatcher = null)
         {
-        }
+            Phase = phase;
+            LogLevelThreshold = logLevelThreshold;
 
-        public void Flush()
-        {
+            // Set default phase matcher to start with current phase
+            PhaseMatcher = phaseMatcher ?? (iLogItem => iLogItem?.Phase != null && iLogItem.Phase.StartsWith(Phase));
```

Comment:
refer to line 38, if phase is null, the item will be added to items, need not check again here?

---
In reply to: [119304710](https://github.com/dotnet/docfx/pull/1712#discussion_r119304710) [](ancestors = 119304710)